mixer/pkg/runtime/routing/builder.go

```diff
@@ -147,6 +148,7 @@ func (b *builder) nextID() uint32 {
 
 func (b *builder) build(snapshot *config.Snapshot) {
 
+rules:
```

Comment:
do we need to refactor this whole block?  I am generally suspicious of using labels and this bit of logic seems to contain a ton of `continue rules` statements across 50+ lines of code. Seems like tracking flow through this section of logic is getting fairly advanced.

Reply:
I agree that this is too long of a function (and in fact, had bugs due to nested `continue`). Let me try to take a stub at refactoring.